brave/browser/brave_browser_context.cc

```diff
@@ -162,6 +175,49 @@ BraveBrowserContext::BraveBrowserContext(
         atom::AtomBrowserContext::From(parent_partition, false));
   }
 
+  bool isolated_storage;
+  if (options.GetBoolean("isolated_storage", &isolated_storage)) {
+    isolated_storage_ = isolated_storage;
+  }
+
+  std::string tor_path;
+  std::string tor_proxy;
+  if (options.GetString("tor_path", &tor_path) &&
+      options.GetString("tor_proxy", &tor_proxy)) {
+    url::Parsed url;
+    url::ParseStandardURL(
+      tor_proxy.c_str(),
+      std::min(tor_proxy.size(),
+               static_cast<size_t>(std::numeric_limits<int>::max())),
+      &url);
+    std::string tor_host, tor_port;
+    if (url.host.is_valid()) {
+      tor_host =
+        std::string(tor_proxy.begin() + url.host.begin,
+                    tor_proxy.begin() + url.host.begin + url.host.len);
+    }
+    if (url.port.is_valid()) {
+      tor_port =
+        std::string(tor_proxy.begin() + url.port.begin,
+                    tor_proxy.begin() + url.port.begin + url.port.len);
+    }
+    base::WaitableEvent* tor_launched =
+      new base::WaitableEvent(
+        base::WaitableEvent::ResetPolicy::MANUAL,
+        base::WaitableEvent::InitialState::NOT_SIGNALED);
+    BrowserThread::PostTask(
+        BrowserThread::PROCESS_LAUNCHER, FROM_HERE,
+        base::Bind(&BraveBrowserContext::LaunchTorProcess,
+                   base::Unretained(this),
+                   tor_launched,
+                   tor_path,
+                   tor_host,
+                   tor_port));
+    tor_launched->Wait();
```

Comment:
Making this synchronous may be OK for now as long as it's basically just a fork/fork/exec, but it seems to me that, in the future, we will want to wait until the tor daemon is actually listening and maybe on the tor network, without blocking the main thread.  To this end, we will need to cause any network requests to wait or to fail until the tor daemon is fully launched.  Do you have any thoughts on this?  (Not necessary to address for now, but we will need to address it eventually.)